presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -163,7 +163,7 @@ public class PrestoS3FileSystem
 {
     private static final Logger log = Logger.get(PrestoS3FileSystem.class);
     private static final PrestoS3FileSystemStats STATS = new PrestoS3FileSystemStats();
-    private static final PrestoS3FileSystemMetricCollector METRIC_COLLECTOR = new PrestoS3FileSystemMetricCollector(STATS);
+    private static PrestoS3FileSystemMetricCollector metricCollector = new PrestoS3FileSystemMetricCollector(STATS);
```

Comment:
```suggestion
    private static RequestMetricCollector metricCollector = new PrestoS3FileSystemMetricCollector(STATS);
```
This would avoid the changes to the `AbstractSdkMetricsCollector`. We could adjust the getter/setter accordingly.